### DISCUSSION TITLE
感谢分享，本地识别结果存在异常

### DISCUSSION THREAD

**Discussion by ManiiXu** [OleehyO/TexTeller#39] [feedback]
首先非常感谢分享该开源项目与模型，目前发现在本地调试中，部分自测的公式图片识别异常，存在复读问题，请帮忙分析下（在该项目提供的部分训练数据集中测试正常）：
![1](https://github.com/OleehyO/TexTeller/assets/37164976/6c9fe291-886b-4b7f-9a0c-046d41d986fc)
识别结果为：
\[\begin{array}{c}\Delta\underline{L}_{\text{\tiny{\it{j}}}\text{\tiny{\it{i}}} \text{\tiny{\it{j}}}\text{\tiny{\it{i}}}\text{\tiny{\it{j}}}\text{\tiny{\it{i}}} \text{\tiny{\it{j}}}\text{\tiny{\it{i}}}\text{\tiny{\it{j}}}\text{\tiny{\it{i}}} \text{\tiny{\it{j}}}\text{\tiny{\it{j}}}\text{\tiny{\it{j}}}\text{\tiny{\it{j}}} \text{\tiny{\it{j}}}\text{\tiny{\it{j}}}\text{\tiny{\it{j}}}\text{\tiny{\it{j}}} \text{\tiny{\it{j}}}\text{\tiny{\it{j}}}\text{\tiny{\it{j}}}\text{\tiny{\it{j}}} \text{\tiny{\it{j}}}\text{\tiny{\it{j}}}\text{\tiny{\it{j}}}\text{\tiny{\it{j}}} \text{\tiny{\it{j}}}\text{\tiny{\it{j}}}\text{\tiny{\it{j}}}\text{\tiny{\it{j}}} \text{\tiny{\it{j}}}\text{\tiny{\it{j}}}\text{\tiny{\it{j}}}\text{\tiny{\it{j}}} \text{\tiny{\it{j}}}\text{\tiny{\it{j}}}\text{\tiny{\it{j}}}\text{\tiny{\it{j}}} \text{\tiny{\it{j}}}\text{\tiny{\it{j}}}\text{\tiny{\it{j}}}\text{\tiny{\it{j}}} \text{\tiny{\it{j}}}\text{\tiny{\it{j}}}\text{\tiny{\it{j}}}\text{\tiny{\it{j}}} \text{\tiny{\it{j}}}\text{\tiny{\it{j}}}\text{\tiny{\it{j}}}\text{\tiny{\it{j}}} \text{\tiny{\it{j}}}\text{\tiny{\it{j}}}\text{\tiny{\it{j}}}\text{\it{ \it{j}}}\text{\tiny{\it{j}}}\text{\tiny{\it{j}}}\text{\it{ \it{j}}}\text{\tiny{\it{j}}}\text{\tiny{\it{j}}}\text{\it{ \it{j}}}\text{\tiny{\it{j}}}\text{\it{j}}}\text{\tiny{\it{j}}}\text{\it{ \it{j}}}\text{\tiny{\it{j}}}\text{\it{ \it{j}}}\text{\tiny{\it{j}}}\text{\it{ \it{j}}}\text{\tiny{\it{j}}}\text{\it{ \it{j}}}\text{\tiny{\it{j}}}\text{\it{ \it{j}}}\text{\tiny{\it{j}}}\text{\it{ \it{j}}}\text{\tiny{\it{j}}}\text{\it{ \it{j}}}\text{\tiny{\it{j}}}\text{\it{ \it{j}}}\text{\tiny{\it{j}}}\text{\it{ \it{j}}}\text{\tiny{\it{j}}}\text{\it{ \it{j}}}\text{\tiny{\it{j}}}\text{\tiny{\it{j}}}\text{\it{ \it{ \it{j}}}\text{\tiny{\it{j}}}\text{\it{ \it{ \it{j}}}\text{\it{ \it{j}}}\text{\tiny{\it{j}}}\text{\it{ \it{ \it{j}}}\text{\tiny{\it{j}}}\text{\it{ \it{ \it{j}}}\text{\it{ \it{ \it{j}}}\text{\tiny{\it{ \it{ \it{j}}}\text{\it{ \it{j}}}\text{\it{ \it{ \it{ \it{j}}}\text{\it{ \it{ \it{ }}\text{ \it{ \it{ }}\text{ \it{ \it{ }}\text{ \it{ \it{ \it{ \it{ \it{ \it{ \it{ \it{ \it{ \it{ \it{ \it{ \it{ \it{ \it{ \it{ \it{ \it{ \it{ \it{ \it{ \it{ \it{ \it{ \bm{ \it}}}\text{ \bm{ \it{ \bm{ \bm{ \bm{ \bm{ \bm{ \bm{ \bm{ \bm{ \bm{ \bm{ \bm{ \bm{ \bm{ \bm{ \bm{ \bm{ \bm{ \bm{ \bm{ \bm{ \bm{ \bm{ \

![2](https://github.com/OleehyO/TexTeller/assets/37164976/ff4919e0-e374-4715-8048-410085c72778)
识别结果为：
\[\bar{b}\bar{=}\frac{\Delta\bar{T_{i}}}{\bar{k_{i}}\bar{=}\bar{k_{i}}\bar{=} \bar{1}\bar{1}\bar{1}\bar{1}\bar{0}\bar{0}\bar{0}\bar{0}\bar{+}\bar{1}\bar{1} \bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{1}\bar{1} \bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0} \bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0} \bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0} \bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0} \bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0} \bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0} \bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\bar{0}\

另外请帮忙确认下目前在huggingface上提供的模型版本已经是TexTeller 2.0吗（基于7.5M数据训练）？

**Comment by ManiiXu** [comment]
以上结果均基于 huggingface.co 的当前模型，与inference.py测试得到。
如果可以的话，请测试下这几张图在您这边的结果，以判断确认是否是模型、代码中有参数异常。

**Comment by OleehyO** [comment]
这个结果是正常的，因为TexTeller的训练数据全来自于arxiv，所以对于非标准latex渲染的公式版式**非常敏感**（比如图二这样的蓝底，word中打出来的公式或不寻常的公式字体）。

现在在用更强的数据增强进行第二轮训练，预计之后才会解决这个问题。

所以目前还是建议只用TexTeller来**识别论文或是katex渲染出来的公式**。
